java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -394,6 +396,244 @@ public static CompletableFuture<GlideString> get(
+     *           matched, returns the length of the first matching array. If <code>path</code> doesn't
+     *           exist or the value at <code>path</code> is not an array, an error is raised.
+     *     </ul>
+     *     If <code>key</code> doesn't exist, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "{\"a\": [1, 2, 3], \"b\": {\"a\": [1, 2], \"c\": {\"a\": 42}}}").get();
+     * var res = Json.arrlen(client, gs("doc"), gs("$")).get();
+     * assert Arrays.equals((Object[]) res, new Object[] { null }); // No array at the root path.
+     * res = Json.arrlen(client, gs("doc"), gs("$.a")).get();
+     * assert Arrays.equals((Object[]) res, new Object[] { 3 }); // Retrieves the length of the array at path $.a.
+     * res = Json.arrlen(client, gs("doc"), gs("$..a")).get();
+     * assert Arrays.equals((Object[]) res, new Object[] { 3, 2, null }); // Retrieves lengths of arrays found at all levels of the path `..a`.
+     * res = Json.arrlen(client, gs("doc"), gs("..a")).get();
+     * assert res == 3; // Legacy path retrieves the first array match at path `..a`.
+     * }</pre>
+     */
+    public static CompletableFuture<Object> arrlen(
+            @NonNull BaseClient client, @NonNull GlideString key, @NonNull GlideString path) {
+        return executeCommand(
+                client, new ArgsBuilder().add(gs(JSON_ARRLEN)).add(key).add(path).toArray());
+    }
+
+    /**
+     * Retrieves the length of the array at the root of the JSON document stored at <code>key</code>.
+     * <br>
+     * Equivalent to {@link #arrlen(BaseClient, String, String)} with <code>path</code> set to <code>
+     * "."</code>.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @return The array length stored at the root of the document. If document root is not an array,
+     *     an error is raised.<br>
+     *     If <code>key</code> doesn't exist, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "[1, 2, true, null, \"tree\"]").get();
+     * var res = Json.arrlen(client, "doc").get();
+     * assert res == 5;
+     * }</pre>
+     */
+    public static CompletableFuture<Long> arrlen(@NonNull BaseClient client, @NonNull String key) {
+        return executeCommand(client, new ArgsBuilder().add(gs(JSON_ARRLEN)).add(key).toArray());
+    }
+
+    /**
+     * Retrieves the length of the array at the root of the JSON document stored at <code>key</code>.
+     * Equivalent to {@link #arrlen(BaseClient, GlideString, GlideString)} with <code>path</code> set
+     * to <code>gs(".")</code>.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @return The array length stored at the root of the document. If document root is not an array,
```

Comment:
@jamesx-improving are we including this signature?  It doesn't seem useful to include a API signature that won't get used.  
We should consider removing this.

Reply:
Which signature?

Reply:
For this command, `path` is indeed optional on all versions of documentations, hence it's hard to justify the removal of this signature (and the GlideString variant of it down below)